packstream/packstream-specification-1.md

```diff
@@ -809,15 +827,67 @@ DateTimeZoneId::Structure(
 )
 ```
 
-- The `seconds` are seconds since the adjusted [Unix epoch](https://en.wikipedia.org/wiki/Epoch_(computing)). This is not [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
-- The `tz_id` is an identifier for a specific time zone, such as `"Europe/Paris"`.
+- The `tz_id` specifies the timezone name as understood
+  by [the timezone database](https://en.wikipedia.org/wiki/Tz_database).
+- The `seconds` elapsed since the [Unix epoch](https://en.wikipedia.org/wiki/Epoch_(computing)), often referred as a
+  Unix timestamp, **added** to the offset derived from the named timezone and specified the point in time.
+- The `nanoseconds` are what remains after the last second of the datetime. The amount of nanoseconds ranges from `0` to
+  `999_999_999` (`_` separator added here and later for clarity).
 
-To convert to UTC:
+For instance, the serialization of the point in time denoted as `1970-01-01T02:15:00+0100[Europe/Paris]` (and `42`
+nanoseconds) can be implemented as follows:
 
+- retrieve the offset of the named timezone for that point in time, here +1 hour, i.e. `3_600` seconds (the resolution
+  is not always defined, read the following `Known Limitations` section to learn more)
+- compute the UTC time, i.e. `1970-01-01T01:15:00Z` (`Z` denotes UTC)
+- compute the difference between that UTC time and the Unix epoch, which is 1h15, i.e. `8_100` seconds.
+- add the resolved offset of 1 hour, i.e. `3_600`, seconds to the above difference, which yields `11_700` (`8_100`+`3_600`)
+
+The resulting `DateTime` instance is therefore as follows:
 ```
-utc_nanoseconds = (seconds * 1000000000) + nanoseconds - get_offset_in_nanoseconds(tz_id)
+{
+  seconds: 11700
+  nanoseconds: 42,
+  tz_id: "Europe/Paris"
+}
 ```
 
+The deserialization of such a `DateTime` structure expectedly happens in reverse:
+
+- retrieve the offset of the named timezone for that point in time, here +1 hour, i.e. `3_600` seconds (the resolution
+  is not always defined, read the following `Known Limitations` to learn more)
+- remove the resolved offset from the `seconds` field, which gives here `8_100`
+- instantiate the idiomatic equivalent of `DateTime` based on that Unix timestamp giving `1970-01-01T01:15:00Z`
+- localize the resulting UTC `DateTime` to the timezone of the specified offset giving `1970-01-01T02:15:00+0100[Europe/Paris]`
+
+#### Known limitations
+
+##### Accuracy
+
+The resolution of offsets for a given time zone name and point in time is bound to the accuracy of the underlying
+[timezone database](https://en.wikipedia.org/wiki/Tz_database). In
+particular, [time zones before 1970](https://en.wikipedia.org/wiki/Tz_database#Data_before_1970) are not as well
+specified.
+
+Moreover, the offset resolution likely occurs both on the Bolt client side and Bolt server side. They each rely on a 
+different timezone database. That could lead to unwanted discrepancies if these copies are not in sync.
```

Comment:
Perhaps make it more explicit: "if these two timezone databases are not in sync".